### ISSUE TITLE
[bug] False positive in react-x/no-context-provider

### ISSUE THREAD

**Issue by kachkaev** [Rel1cx/eslint-react#991]
👋  @Rel1cx!

### Describe the bug

[react-x/no-context-provider](https://eslint-react.xyz/docs/rules/no-context-provider) tiggers on symbols named `Provider` which are imported from third-party libs.

### Reproduction

Example:

```js
import { Provider } from "jotai";

function Component() {
  return <Provider>hello world</Provider>;
}
```

ESLint output:

```
In React 19, you can render '<Context>' as a provider instead of '<Context.Provider>'.
eslint@eslint-react/no-context-provider
```

### Expected behavior

No error

### Platform and versions

```Text
@eslint-react/eslint-plugin@1.32.1
```

### Additional context

Likely cased by https://github.com/Rel1cx/eslint-react/pull/985. The above code snippet works without errors in `1.32.0`.

Changing code like this mitigates the error:

```js
import { Provider as JotaiProvider } from "jotai";

function Component() {
  return <JotaiProvider>hello world</JotaiProvider>;
}
```

**Comment by Rel1cx** [comment]
Hi @kachkaev! The example above is very common. I will fix it.
